syntax = "proto3";

message EmptyMessage { }

message PingSalesRecordsResponse{

    string ack = 1;

}

message SalesRecordResponse{
    string data = 1;
}

message SalesRecordRequest{
    string region = 1;
    string item_type = 2;
    string units_sold = 3;
    string unit_price = 4;
    string unit_cost = 5;
    string source = 6;
}

service SalesRecord {
    rpc PingSalesRecords(EmptyMessage) returns (PingSalesRecordsResponse);
    rpc SendSalesRecord(SalesRecordRequest) returns (SalesRecordResponse);
}
